### PR TITLE
Fix model creation when redirect is required.

### PR DESCRIPTION
**Pull request by fabricematrat**
If you connect to jimm, when trying to add a model with DF the redirection to the model API is not handle properly so we login after the commit. This means we try to access the model api before the login so it fails.

### CODE REVIEW

**Comment by fabricematrat** [comment]
QA:

Connect your GUI to jimm on staging and try the deployment flow with new model existing model and then try to see if it breaks in oher scenario.
Check that the model name is correct that you can deploy switch etc ...

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6032/
Test FAILed.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -920,7 +920,8 @@ YUI.add('juju-gui', function(Y) {
         // Switch to the redirected model.
         this.switchEnv(this.createSocketURL(
           this.get('socketTemplate'),
-          this.get('modelUUID'), publicHost.value, publicHost.port));
+          this.get('modelUUID'), publicHost.value, publicHost.port), null,
+          null, null, true, false);
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/app.js]: Would it be worth adding a comment here to explain the state these nulls and bools are setting?

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -1484,6 +1484,12 @@ YUI.add('juju-view-utils', function(Y) {
     });
     // Update the model name. The onEnvironmentNameChange in app.js method will
     // update the name correctly accross components.
+    var cb = function(params) {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/utils.js]: Could you call this function something more descriptive please?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6034/
Test PASSed.

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (fabricematrat) [inline jujugui/static/gui/src/app/app.js]: done.

**Comment by bac** [comment]
Tried QA using gisf-qa-tools and @jaycee's blues-browser PR 1410 per @fabricematrat's instructions.

Following the detailed instructions in PR1410 I was able to get a GUI. When trying to deploy I was able to select AWS but then the remainder of the page was greyed out. No way to pick credentials or deploy.

**Comment by hatched** [comment]
It turned out that the issue @bac found was an issue generating the aws creds, switching to gce and everything worked as expected. The other setups will be qa'd as well.

**Comment by hatched** [comment]
The model name issue is unrelated and I've been working on a fix.

**Comment by bac** [comment]
QA-ok for GIJOE and charm deploy.

**Comment by hatched** [comment]
Shipping this and I also created a follow-up task for 2.2.3 to implement this using the alternative approach. Thanks @fabricematrat 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**